malloc/malloc.c

```diff
@@ -43,7 +59,31 @@ void *mem_copy(void *dest, const void *src, int n)
 */
 void *resize_memory(void *ptr, int old_size, int new_size)
 {
+    if (new_size == 0){
+        free(ptr);
+        return NULL;
+    }
+
+    else if (!ptr) {
+        return malloc(new_size);
+    }
+
+    else if (old_size == new_size) {
+        return ptr;
+    }
+
+    void *new_block = malloc(new_size);
+
+    if (new_size < old_size) {
+        mem_copy(new_block, ptr, new_size);
+    }
+
+    else {
+        mem_copy(new_block, ptr, old_size);
+    }
```

Comment:
The test for resize memory isn't passing. The tests are checking that if the existing memory has a string and you resize it to be less than it's original size, the string in the resized memory is properly truncated (i.e. is properly terminated with a '\0'.